rtengine/boxblur.h

```diff
@@ -312,6 +313,222 @@ template<class T, class A> void boxblur (T** src, A** dst, T* buffer, int radx,
 
 }
 
+inline void boxblur (float** src, float** dst, int radius, int W, int H, bool multiThread)
+{
+    //box blur using rowbuffers and linebuffers instead of a full size buffer
+
+    if (radius == 0) {
+        if (src != dst) {
+#ifdef _OPENMP
+            #pragma omp parallel for if (multiThread)
+#endif
+
+            for (int row = 0; row < H; row++) {
+                for (int col = 0; col < W; col++) {
+                    dst[row][col] = src[row][col];
+                }
+            }
+        }
+        return;
+    }
+
+    constexpr int numCols = 8; // process numCols columns at once for better usage of L1 cpu cache
+#ifdef _OPENMP
+    #pragma omp parallel if (multiThread)
+#endif
+    {
+        float* const buffer = new float[std::max(W, 8 * H)];
+        //horizontal blur
+        float* const lineBuffer = buffer;
+#ifdef _OPENMP
+        #pragma omp for
+#endif
+        for (int row = 0; row < H; row++) {
+            float len = radius + 1;
+            float tempval = src[row][0];
+            lineBuffer[0] = tempval;
+            for (int j = 1; j <= radius; j++) {
+                tempval += src[row][j];
+            }
+
+            tempval /= len;
+            dst[row][0] = tempval;
+
+            for (int col = 1; col <= radius; col++) {
+                lineBuffer[col] = src[row][col];
+                dst[row][col] = tempval = (tempval * len + src[row][col + radius]) / (len + 1);
+                len ++;
```

Comment:
`++len;`

I for one refrain from multiple assignments per line (`dst[row][col] = tempval =...`) but it's readable here...

Reply:
Both fixed.